### PR TITLE
Show broker_connection_retry_on_startup warning only if it evaluates as False

### LINKED ISSUE #9225: Confusing warning about `broker_connection_retry_on_startup`
# Checklist

- [x] I have checked the [issues list](https://github.com/celery/celery/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3A%22Category%3A+Documentation%22+)
  for similar or identical bug reports.
- [x] I have checked the [pull requests list](https://github.com/celery/celery/pulls?q=is%3Apr+label%3A%22Category%3A+Documentation%22)
  for existing proposed fixes.
- [x] I have checked the [commit log](https://github.com/celery/celery/commits/main)
  to find out if the bug was already fixed in the main branch.
- [x] I have included all related issues and possible duplicate issues in this issue
      (If there are none, check this box anyway).

#### Related Issues

- None

#### Possible Duplicates

- None

# Description

We just update Celery and we get this warning:

```
python-3.11/lib/python3.11/site-packages/celery/worker/consumer/consumer.py:508: 
CPendingDeprecationWarning: The broker_connection_retry configuration setting will no longer determine
whether broker connection retries are made during startup in Celery 6.0 and above.
If you wish to retain the existing behavior for retrying connections on startup,
you should set broker_connection_retry_on_startup to True.
```

Since in our configuration we didn't touch either (`broker_connection_retry` nor `broker_connection_retry_on_startup`), it took me some time to understand what is going on.

Also, from the message it looks like that the default for `broker_connection_retry_on_startup` will be `False`, but the [docs](https://docs.celeryq.dev/en/stable/userguide/configuration.html#broker-connection-retry-on-startup) specify that's enabled by default.

# Suggestions

Assuming that the default in Celery 6 will be to retry every time (both `True` by default), I think that the deprecation warning should not appear if `broker_connection_retry is True`, as no change would result from the deprecation going into action.

To recap, that's my proposition:
- `broker_connection_retry` unset or `True`, `broker_connection_retry_on_startup` unset: no warning (everything evaluated as `True`) <= suggested change
- `broker_connection_retry=False`, `broker_connection_retry_on_startup` unset: warning as it is now
- both `broker_connection_retry` and `broker_connection_retry_on_startup` set: no warning, as it is now

Let me know what are your thoughts, I can prepare a little PR if you agree.

### PR DESCRIPTION
**Pull request by serl**
## Description

Fixes #9225. As said in the issue, the warning is confusing if the user is sticking to the defaults already (as there's nothing to do). With this PR, the warning is shown only if the configuration would actually change with the 6.0 update.

### CODE REVIEW

**Review by Nusnus — CHANGES_REQUESTED** [review]
Please fix failing tests, it appears to be a real bug.

**Comment by auvipy** [comment]
the test seems passing, isn't it? what is actually failing?

**Comment by Nusnus** [comment]
> the test seems passing, isn't it? what is actually failing?

Ohhh.. now I see where I commented this hahah
It was a mistake. I typed it from my phone in the wrong PR 😅

**Review by auvipy — APPROVED** [review]

File: celery/worker/consumer/consumer.py [hunk 1]
```diff
@@ -505,13 +505,14 @@ def _error_handler(exc, interval, next_step=CONNECTION_RETRY_STEP):
...
-                CPendingDeprecationWarning(
-                    f"The broker_connection_retry configuration setting will no longer determine\n"
-                    f"whether broker connection retries are made during startup in Celery 6.0 and above.\n"
-                    f"If you wish to retain the existing behavior for retrying connections on startup,\n"
-                    f"you should set broker_connection_retry_on_startup to {self.app.conf.broker_connection_retry}.")
-            )
+            if retry_disabled:
+                warnings.warn(
+                    CPendingDeprecationWarning(
+                        "The broker_connection_retry configuration setting will no longer determine\n"
```
Comment (auvipy) [inline celery/worker/consumer/consumer.py:511]: should we keep using the f"string" here as before?

Comment (serl) [reply celery/worker/consumer/consumer.py:511]: You mean with the `{self.app.conf.broker_connection_retry}`? I don't think so, given that I also changed "retain the existing behavior" to "refrain from retrying connections".
Actually, I copied the warning in the [doc](https://docs.celeryq.dev/en/stable/userguide/configuration.html#broker-connection-retry), I think having the exact same words here and there make a lot of sense

(Or if you were just talking about the `f` before the `"`, the linter complained about it being useless given that there's no variables inside anymore)

**Review by thedrow — APPROVED** [review]
LGTM

**Pull request merged**